node/src/database/archive.rs

```diff
@@ -0,0 +1,316 @@
+
+        info!(
+            "Archived events from block {} with height {}",
+            util::truncate_string(&hex_block_hash),
+            block_height
+        );
+
+        Ok(())
+    }
+
+    /// Fetch the list of all vm events from the block of the given height.
+    async fn fetch_vm_events(
+        &self,
+        block_height: u64,
+    ) -> Result<Vec<ContractTxEvent>> {
+        let block_height: i64 = block_height as i64;
+
+        let mut conn = self.archive_db.acquire().await?;
+
+        let events = sqlx::query!(
+            r#"SELECT json_contract_events FROM archive WHERE block_height = ?"#,
+            block_height
+        ).fetch_one(&mut *conn).await?;
+
+        // convert the json string to a vector of ContractTxEvent and return it
+        Ok(serde_json::from_str(&events.json_contract_events)?)
+    }
+
+    /// Mark the block of the given height and hash as finalized in the archive.
+    async fn mark_block_finalized(
+        &self,
+        block_height: u64,
+        hex_block_hash: String,
+    ) -> Result<()> {
+        let block_height: i64 = block_height as i64;
+
+        let mut conn = self.archive_db.acquire().await?;
+
+        // Set finalized to true for the block with the given hash
+        // Return the block height
+        let r = sqlx::query!(
+            r#"UPDATE archive SET finalized = 1 WHERE block_hash = ?
+            RETURNING block_height
+            "#,
+            hex_block_hash
+        )
+        .fetch_one(&mut *conn)
+        .await?;
+
+        info!(
+            "Marked block {} as finalized at height {}. After {} blocks",
+            util::truncate_string(&hex_block_hash),
+            block_height,
+            block_height - r.block_height
+        );
+
+        Ok(())
+    }
+
+    /// Remove the block of the given height and hash from the archive.
+    async fn remove_deleted_block(
+        &self,
+        block_height: u64,
+        hex_block_hash: String,
+    ) -> Result<bool> {
+        let block_height: i64 = block_height as i64;
+
+        let mut conn = self.archive_db.acquire().await?;
+
+        let r = sqlx::query!(
+            r#"DELETE FROM archive WHERE block_hash = ? AND (finalized IS NULL OR finalized = 0)
+            RETURNING block_height
+            "#,
+            hex_block_hash
+        )
+        .fetch_optional(&mut *conn)
+        .await?;
+
+        if let Some(r) = r {
+            info!(
+                "Deleted events from block {} with block height: {} at height {}",
+                util::truncate_string(&hex_block_hash),
+                r.block_height,
+                block_height
+            );
+            Ok(true)
+        } else {
+            warn!(
+                "Trying to delete Block {} which is finalized or does not exist in the archive",
+                util::truncate_string(&hex_block_hash)
+            );
+            Ok(false)
+        }
+    }
+}
+
+mod util {
```

Comment:
I would avoid the use of something general like "util" And @ZER0 has also emphasied it multiple times, the module can contain anything its name is not specific

On the other hand, where is this used? The `truncate_string` function can panic and is `pub` there's no doc comment that it can panic, it doesnt return an error type either

Reply:
Let's discuss this in the next PR when the archive storage functionality is finalized